docs/proposals/Drawer/README.md

```diff
@@ -0,0 +1,70 @@
+# Drawer
+
+The `<Drawer>` component will provide a dismissible side panel.
+
+## Design Patterns
+
+`<Drawer>` It will allow the user to view related content to their main task.
+Unlike a `<Modal>`, it will take full advantage of the available viewport
+height. This will make it ideal for long lists and other scrolling content.
+
+Some scenarios for a `<Drawer>` include the following:
+
+- Show a list of notes beside a job detail view
+- Show a work request beside a new quote form
+- Show a converted quote beside an active job
+- Show a list of insight cards beside a dashboard
+
+There are also some outstanding questions:
+
+- Should the drawer be full-height of the viewport or just the parent container?
+- How should the drawer animate when it opens and closed?
+- Should we include an option for left-hand and right-hand drawers?
+- Do all drawers need to be dismissible? What about an always-open version?
+- Could the unscheduled visits accordion on the calendar turn into a drawer?
+
+## Wireframe
+
+https://www.figma.com/file/wKlxfdTZlrffPrPiWpEOEB/Insightful-Dashboard-Components?node-id=0%3A1
+
+## Interface
+
+```
+<Drawer title="Request" subtitle="2020-02-10">
+  <Content>
+    Request goes here...
+  </Content>
+</Drawer>
+```
+
+## Props Table
+
+| name      | type    | default | description                                             |
+| --------- | ------- | ------- | ------------------------------------------------------- |
+| title     | String  | -       | The title of the drawer                                 |
+| subtitle  | String  | -       | The subtitle of the drawer                              |
+| overlay   | String  | -       | If the drawer overlays the main content or displaces it |
```

Comment:
I think that if `overlay` is true this very much becomes a modal. In those cases would a modal be a better fit?

Reply:
I wonder if this is a key difference between a 'side modal' and a 'side drawer' that would keep us from wanting to use it, even in overlay use cases. This is a key aspect of Modal: "It also prevent users from interacting with the rest of the application until a specific action is taken".   Seems like drawer use cases might be less flow-interrupting, and should probably support easy dismissal (escape key? click out? swipe away?), especially if they're overlaying other content.